### DISCUSSION TITLE
Leaving button options are not translated into Portuguese

### DISCUSSION THREAD

**Discussion by maverick74** [lxqt/lxqt#2090] [Q&A]
I don't know if this is Lubuntu Specific Or LXQt specific but:
When i try to shutdown the computer the dialog window that pops up is only partially translated.

![dialog box](https://user-images.githubusercontent.com/14912461/137155301-ee932880-e9ad-490d-b75e-bbda90c2c214.jpg)

"Yes" / "No" buttons remain in English (from my experience with lubuntu it has always been this way)

* Distribution & Version: Lubuntu 21.10 + LXQt 0.17

**Comment by tsujan** [comment]
Install `qt5-translations` (find its package name in Ubuntu).

**Comment by maverick74** [comment]
hi @tsujan 

It's apparently called `qttranslations5-l10n`, but it was already installed by default... any other clues?

**Comment by tsujan** [comment]
Those strings may have not been translated in your language by Qt or there may be a problem in Ubuntu; I don't know. What I know is that we don't use the strings "Yes" and "No" in https://github.com/lxqt/liblxqt/blob/master/lxqtpowermanager.cpp but only Qt's standard buttons `QMessageBox::Yes` and `QMessageBox::No`. Therefore, Qt is responsible for their translations.

@stefonarch may know about it.

**Comment by tsujan** [comment]
@maverick74
I don't install LXQt translations except for a few apps; lximage-qt is among them. I noticed that if I start it in Portuguese (with `LANGUAGE=pt lximage-qt`), the OK and Cancel buttons of its Preferences dialog won't be translated but if I start it in Spanish, French, Italian or German, they'll be translated.

So, my guess is that Qt's translations into Portuguese are incomplete.

**Comment by maverick74** [comment]
@tsujan 
is there a place i can help translate it (qt's weblate or something like it)?

**Comment by luis-pereira** [comment]
I tried it on Arch and got the same result. Definitely something is wrong with Qt Portuguese translations.

**Comment by tsujan** [comment]
@maverick74

The following screenshots of `kvantumpreview` (which is completely independent of LXQt) confirm my suspicion. This one is in Portuguese:

![pt](https://user-images.githubusercontent.com/981076/137185166-3a322f74-ff9a-409b-b0a1-595ad395d416.png)

This in Spanish:

![es](https://user-images.githubusercontent.com/981076/137185252-4e1bca35-f9fa-4ab9-ad27-aa9077309ccc.png)

> is there a place i can help translate it (qt's weblate or something like it)?

My knowledge in that area is next to zero ;).

However, I convert your report to a Discussion for others to find it easily.

**Comment by luis-pereira** [comment]
Will take a look. Portuguese is my native tongue.

**Reply by maverick74** [reply]
yeah, mine too (pt-pt). If i can be of any assistance in the translations i'll be glad to help (it's not the first time on the job)

**Reply by tsujan** [reply]
> If i can be of any assistance in the translations

The translation needs to be done in Qt itself; LXQt can't do anything about it. However, we still don't know if Qt's issue is in its translations (the missing ones) or somewhere in its code.

**Reply by maverick74** [reply]
@tsujan I see. Thank you!

I'm not all that familiar with how Qt works. 
Do they accept community translations? if yes, where? 
And how about filling bug reports? is there a place for it?

@tsujan  & @luis-pereira if you find something, if you report a bug or if you fix the problem in Qt,  please,say something back here :)

**Reply by tsujan** [reply]
> Do they accept community translations?

My knowledge of how they accept translations is like yours ;) Maybe, someone could tell us how it's done.

> And how about filling bug reports?

Bug reports should be made at [Qt's bug tracker](https://bugreports.qt.io). Since @luis-pereira's native language is Portuguese and because Qt devs usually want sample codes, I think his report will attract more attention. But, as he said above, he needs to investigate the problem first, and that might take time.

**Reply by tsujan** [reply]
I found this: https://bugreports.qt.io/browse/QTBUG-39229

The reason they gave for closing it was this:

> it's closed because it makes no sense to track it.
> translations are community contributions, and they
> come when they come, and when they do, they usually do en masse.

However, there's also this quotation:

> The French, German, and Simplified Chinese translation files are
> supported by Qt. All other translations are contributed by
> third parties.

I'm confused!

So, I guess, another bug report might be closed as well :(

**Comment by luis-pereira** [comment]
The issue is definitely in Qt. `pt_PT` doesn't have updated translations.
 A bug report won't get anything done. Contributing translations is what is needed.
 
All translations are contributed by the Qt community.See the [5.15-tr README ](https://code.qt.io/cgit/qt/qttranslations.git/tree/translations/README?h=5.15-tr) and [qttranslations-6.2.1](https://code.qt.io/cgit/qt/qttranslations.git/tree/translations/README?h=6.2.1).

`pt_BR` has "recently" received contributions. It's just a matter of time for it to get to distributions.  `pt_PT` (formerly pt) isn't updated since the end of  Qt4.
I tried to update translations but I got 750 unfinished translations for Qt5 qtbase and 1329 for Qt6 qtbase. I don't have the time to do it.

**Reply by maverick74** [reply]
Hi @luis-pereira 
Is there somekind of weblate system in place, where we can just sign in and start translating?
If there is i think can get someone to do it!

And why did they change pt > pt-PT? that does not make any sense at all!!!

**Comment by luis-pereira** [comment]
@maverick74  [Qt localization](https://wiki.qt.io/Qt_Localization) page doesn't refer to a web interface.
I really don't know why. Maybe because pt_BR is being maintained.

**Reply by maverick74** [reply]
@luis-pereira That's weird. I see they use transifex.

If there was a "pt" branch i (or someone else around me) could start the translations. 
I don't mean we'll translate the whole thing, but at least some of it as long as we can...

**Reply by luis-pereira** [reply]
It seems that there will be a pt_PT not pt. I missed the Transifex stuff. The submission process has to be done via Gerrit and there's no Qt6 on transifex (weird).
Will come back online on Nov 2.

**Reply by maverick74** [reply]
@luis-pereira there's still no pt_PT on transifex...

Do you have any news on this?
